### ISSUE TITLE
unclosed file warning for signac find and view

### ISSUE THREAD

**Issue by atravitz** [glotzerlab/signac#496]
### Description

I came across this error in the `signac-examples` notebook `signac_105_Command_Line_Interface.ipynb`

### To reproduce

```
signac view -i index.txt ./view
```

or 
```
signac find -i index.txt '{"p": 0.1}'
```

### Error output
```
Reading index from file 'index.txt'...
/Users/atravitz/miniconda3/envs/alyssa/lib/python3.8/site-packages/signac/contrib/collection.py:457: ResourceWarning: unclosed file <_io.TextIOWrapper name='index.txt' mode='r' encoding='UTF-8'>
  for doc in docs:
ResourceWarning: Enable tracemalloc to get the object allocation traceback
```

### System configuration

Please complete the following information:

 - Operating System [e.g. macOS]:
 - Version of Python [e.g. 3.7]:
 - Version of signac [e.g. 1.0]:

Or copy & paste the output of: `python -c 'import platform; print(platform.platform()); import sys; print(sys.version); import signac; print(signac.__version__)'`

**Comment by vyasr** [comment]
@atravitz From a brief look at the code, my guess is that the index file is being read without a context manager to close it here: https://github.com/glotzerlab/signac/blob/master/signac/__main__.py#L192

```
def _read_index(project, fn_index=None):
    if fn_index is not None:
        _print_err(f"Reading index from file '{fn_index}'...")
        file_descriptor = open(fn_index)
        return (json.loads(line) for line in file_descriptor)
```

You can try to just replace that with `with open(fn_index) as file_descriptor:` and see if that fixes it. If so, I'd vote to just close this issue since the indexing module is deprecated and will be removed soon enough anyway.

**Comment by bdice** [comment]
@vyasr We ought to fix this even though the module is deprecated, because a warning shows up in our tutorial notebooks. (Whether indexing should be in the tutorials at all is a separate question, but I think not.)

**Comment by vyasr** [comment]
Yes, I'm suggesting to remove the offending line from the tutorial entirely. Even if you fix this issue, deprecation warnings will show up (leaving aside discussion of automatic DeprecationWarning suppression, our best practices/examples should not use deprecated code).

**Comment by mikemhenry** [comment]
We should leave this issue open (so other people can find it) and close when we remove the model (which I think we can do in 1.7?) since that will "fix" the issue.

**Comment by vyasr** [comment]
The module won't be removed until 2.0 since indexing-based APIs (and CLIs) are public. I went ahead and just fixed the problem in #498. When we get around to doing more deprecations for 1.7 we will need to make sure index function arguments are deprecated everywhere that they should be.
